File: Fundamentals_of_Software_Engineering/front/lk/common/src/commonMain/kotlin/org/suai/common/LoginRepository.java
package org.suai.common;
import org.suai.common.model.loginModel;

import java.io.DataOutputStream;
import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.URL;
import java.nio.charset.StandardCharsets;
import java.util.Arrays;

public class LoginRepository {
    public static Response getResponse(loginModel data){
        try {
            return new Response(sendLoginData(data));
        } catch (IOException e) {
            e.printStackTrace();
            return new Response("Throw exception");
        }
    }

    public static String sendLoginData(loginModel data) throws IOException {
        String request = "http://www.lichuhasite.somee.com/Login/Add";
        String urlParameters = "";

        byte[] postData = urlParameters.getBytes(StandardCharsets.UTF_8);
        URL url = new URL(request);
        HttpURLConnection conn= (HttpURLConnection) url.openConnection();
        conn.setDoOutput(true);
        conn.setInstanceFollowRedirects(false);
        conn.setRequestMethod("POST");
        conn.setRequestProperty("Content-Type", "application/json");
        conn.setRequestProperty("charset", "utf-8");
        conn.setRequestProperty("Content-Length", Integer.toString(postData.length));
        conn.setUseCaches(false);

        try(DataOutputStream wr = new DataOutputStream(conn.getOutputStream())) {
            wr.write(postData);
            return wr.toString();
        }catch (Exception e){
            return "";
        }
    }
}

File: Fundamentals_of_Software_Engineering/front/lk/common/src/commonMain/kotlin/org/suai/common/LoginRepository.kt
package org.suai.common

import okhttp3.MediaType.Companion.toMediaTypeOrNull
import okhttp3.RequestBody.Companion.toRequestBody
import okhttp3.*
import java.io.IOException
import org.suai.common.model.loginModel
import mu.KotlinLogging
import org.suai.common.net.Net
import org.suai.common.net.REQUEST

object LoginRepository {
    fun getResponse(data: loginModel): Response {
        return try {
            Response(sendLoginData(data))
        } catch (e: Exception) {
            e.printStackTrace()
            Response("Throw exception")
        }
    }

    @Throws(IOException::class, InterruptedException::class)
    fun sendLoginData(data: loginModel) = Net.request(REQUEST.POST, "Login/Auth", """{"login":"${data.login}", "password":"${data.password}"}""")
}